WordPress/gutenberg
# Storybook: Add global CSS switcher

Closes #42717
Fixes #42414

## What?

Implements a global CSS switcher add-on so we can test our component styles in different CSS environments.

## Why?

Recently we've made some unexpected discoveries (#37555, #42414, #42415) about the CSS contexts that wp-components exist in. I was also bitten by this when working on #42000, when I wrote CSS for a component that looked good in Storybook but completely broke down when rendered in wp-admin, due to some broadly scoped styles in a globally loaded WP stylesheet.

We need to ensure that our components have sufficient styles to work in a non-WP context. At the same time, we need to ensure that components have styles that are defensive enough to hold up in a (sometimes adverse) WP context.

The purpose of this Storybook add-on is to help us easily switch between WP and non-WP CSS environments.

## How?

First of all, we fix the style leakage issue in the Block Editor Playground. The reset.css that is meant for the Playground will now only be loaded in the Playground story.

Then, we implement a Storybook add-on where we can configure some presets to load and unload CSS files dynamically. I've set up three presets to begin with: `none`, `basic`, and `wordpress`. We can tweak these as we discover new surprises or our testing needs change.

## Testing Instructions

1. `npm run storybook:dev`
2. Pick a story (e.g. FontSizePicker) and try switching between CSS presets.
3. Inspect an `input` element for example, and see how styles from forms.css is affecting it when choosing the WordPress preset.

## Screenshots or screencast 

https://user-images.githubusercontent.com/555336/181296950-bae7bd71-1218-42e7-a80b-97fc7f8fcb65.mp4
